main.go

```diff
@@ -152,16 +163,13 @@ func getCacheDownloadURL(cacheAPIURL string) (string, error) {
 	return respModel.DownloadURL, nil
 }
 
-// parseStackID reads the stack id from the given json bytes.
-func parseStackID(b []byte) (string, error) {
-	type ArchiveInfo struct {
-		StackID string `json:"stack_id,omitempty"`
+// parseArchiveInfo reads the stack id and architecture from the given json bytes.
+func parseArchiveInfo(b []byte) (archiveInfo, error) {
```

Comment:
NOTE: i was thinking about this:

```
// parseArchiveInfo reads the stack id and architecture from the given json bytes.
func parseArchiveInfo(b []byte) (info archiveInfo, err error) {
	err = json.Unmarshal(b, &info)
	return
}
```

The only difference is in the case of some errors `info` is not an empty struct.